pkg/otlp/metrics/metrics_translator.go

```diff
@@ -51,9 +55,10 @@ func (*noSourceProvider) Source(context.Context) (source.Source, error) {
 
 // Translator is a metrics translator.
 type Translator struct {
-	prevPts *ttlCache
-	logger  *zap.Logger
-	cfg     translatorConfig
+	prevPts        *ttlCache
+	logger         *zap.Logger
+	missingSources otelmetric.Int64Counter
```

Comment:
I'm not sure how many more metrics we will add, but my concern with adding this as a field in the `Translator` struct is then each time we want to add a new metric we need to add a field to this struct.

It may be easier to have a separate struct that stores all our instruments:
```
type Translator struct {
	instruments instruments
}

type instruments struct {
	missingSources otelmetric.Int64Counter
	othermetric    otelmetric.Float64Histogram
}
```

What do you think ?

Reply:
Makes sense, changed it!